### PR TITLE
feat: add new API to advertise accelerator plugin's metadata

### LINKED ISSUE #1085: Let accelerator plugins provide richer resource slot metadata
Currently `/config/resource-slots` API only provides the resource slot names and their types (`COUNT`, `BYTES`, `UNIQUE`).

Let's add more information for extensibility of webui/control-panel:

- Human-friendly resource slot name (to be shown in the UI)
- The desired number format (fractions, thousand delimeters, etc.)
- The desired UI resource icon type
- Description

For backward compatibility of existing accelerator plugins, let's fill up these information in the manager for known resource slots and use the information if the (future) plugin provides.

### PR DESCRIPTION
**Pull request by kyujin-cho**
This resolves #1085.

### CODE REVIEW

File: src/ai/backend/manager/api/etcd.py [hunk 1]
```diff
@@ -21,13 +24,57 @@
...
+    log.info("ETCD.GET_RESOURCE_METADATA ()")
+    root_ctx: RootContext = request.app["_root.context"]
+    available_slot_metadata: Dict[str, Any] = {}
+
+    async with root_ctx.db.begin_readonly() as conn:
+        query = sa.select([agents.c.compute_plugins]).select_from(agents).where()
+        result = await conn.execute(query)
+        for row in result:
+            for key, value in row["compute_plugins"].items():
+                if _data := value.get("metadata", KNOWN_SLOT_METADATA.get(key)):
```
Comment (achimnol) [inline src/ai/backend/manager/api/etcd.py:91]: What happens if there is no metadata in both the value and `KNOWN_SLOT_METADATA`?
Let's add a little more friendly error description.

File: src/ai/backend/manager/api/etcd.py [hunk 2]
```diff
@@ -21,13 +24,57 @@
 log = BraceStyleAdapter(logging.getLogger(__spec__.name))  # type: ignore[name-defined]
 
 
+KNOWN_SLOT_METADATA: Mapping[str, AcceleratorMetadata] = {
```
Comment (achimnol) [inline src/ai/backend/manager/api/etcd.py:27]: Let's add TPU and ROCM defaults.

**Review by achimnol — CHANGES_REQUESTED** [review]

File: src/ai/backend/manager/api/etcd.py (hunk 1 above)
Comment (kyujin-cho) [reply src/ai/backend/manager/api/etcd.py:91]: I've ignored raising error for slot types without metadata intentionally. Added a brief description about these as a block of comment.

Comment (achimnol) [reply src/ai/backend/manager/api/etcd.py:91]: Um, `dict.get()`'s default second argument is `None`, not `{}`.

**Pull request merged**